### ISSUE TITLE
Add internal logging system

### ISSUE THREAD

**Issue by jasondellaluce** [falcosecurity/plugin-sdk-go#24]
**Motivation**
Currently, the SDK does not have an embedded logging system. This could be useful for debugging plugins during the development phase and in many other use cases. Currently this is attained manually in each plugin ([see an example]( https://github.com/falcosecurity/plugins/blob/019437ed9d427ba513555dc9ad8a2d5a3415bec0/plugins/cloudtrail/cloudtrail.go#L200)).

**Feature**
Add some standard Go logging system in the SDK, such as https://pkg.go.dev/log. Plugins must be able to enable/disable and configure the logger programmatically.

The logging system could be bundled in the base level `sdk` package, but we can also consider creating an single-reposibility `log` package, just like we have `ptr` and `cgo`. Then, each internal part of the SDK should rely on that to log information with different severity level. For instance, each prebuilt C symbol should log meaningful information, and constructs such as `EventWriters` can potentially log something too.

**Alternatives**
Leave things as they are, meaning that user each plugin should implement their logging measures. I don't like this option, especially because it makes not possible for the SDK to expose some of its internal process, which can be meaningful for debugging purposes.

**Comment by GLVSKiriti** [comment]
@jasondellaluce I am also interested in this issue! I like to work on this if it is available

**Comment by jasondellaluce** [comment]
Yeah definitely, let me assign you to this one. Note that the latest plugin API provides logging primitives -- take a look at the work we did for the SDK C++: https://github.com/falcosecurity/plugin-sdk-cpp/pull/35

/assign GLVSKiriti

**Comment by leogr** [comment]
Any update on this? :thinking:
